src/overlays/actors/ovl_Dm_Gm/z_dm_gm.c

```diff
@@ -31,37 +33,336 @@ const ActorInit Dm_Gm_InitVars = {
+    s8 objectIndex = this->actor.objBankIndex;
+    s8 objectIndex2;
+    s32 ret = false;
+
+    if (arg2 < 2) {
+        objectIndex2 = this->actor.objBankIndex;
+    } else {
+        objectIndex2 = this->unk_2AC;
+    }
+
+    if ((objectIndex2 >= 0) && (arg2 != this->unk_2C8)) {
+        gSegments[6] = PHYSICAL_TO_VIRTUAL2(globalCtx->objectCtx.status[objectIndex2].segment);
+        this->unk_2C8 = arg2;
+        ret = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimations, arg2);
+        gSegments[6] = PHYSICAL_TO_VIRTUAL2(globalCtx->objectCtx.status[objectIndex].segment);
+    }
+    return ret;
+}
+
+void func_80C24504(DmGm* this) {
+    if (DECR(this->unk_2BA) == 0) {
+        this->unk_2B8++;
+        if (this->unk_2B8 >= 4) {
+            this->unk_2BA = Rand_S16Offset(30, 30);
+            this->unk_2B8 = 0;
+        }
+    }
+}
+
+s32 func_80C2457C(DmGm* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f sp40;
+    Vec3f sp34;
+    s16 sp32;
+
+    Math_Vec3f_Copy(&sp40, &this->unk_2B4->world.pos);
+    Math_Vec3f_Copy(&sp34, &this->actor.world.pos);
+    sp32 = Math_Vec3f_Yaw(&sp34, &sp40);
+
+    Math_ApproachS(&this->unk_2C0, (sp32 - this->unk_2C4) - this->actor.shape.rot.y, 4, 0x2AA8);
+    this->unk_2C0 = CLAMP(this->unk_2C0, -0x1FFE, 0x1FFE);
+
+    Math_ApproachS(&this->unk_2C4, (sp32 - this->unk_2C0) - this->actor.shape.rot.y, 4, 0x2AA8);
+    this->unk_2C4 = CLAMP(this->unk_2C4, -0x1C70, 0x1C70);
+
+    if (this->unk_2B4->id == ACTOR_PLAYER) {
+        sp40.y = ((Player*)this->unk_2B4)->bodyPartsPos[7].y + 3.0f;
+    } else {
+        Math_Vec3f_Copy(&sp40, &this->unk_2B4->focus.pos);
+    }
+
+    Math_Vec3f_Copy(&sp34, &this->actor.focus.pos);
+    Math_ApproachS(&this->unk_2BE, Math_Vec3f_Pitch(&sp34, &sp40) - this->unk_2C2, 4, 0x2AA8);
+    this->unk_2BE = CLAMP(this->unk_2BE, -0x1C70, 0x1C70);
+
+    Math_ApproachS(&this->unk_2C2, Math_Vec3f_Pitch(&sp34, &sp40) - this->unk_2BE, 4, 0x2AA8);
+    this->unk_2C2 = CLAMP(this->unk_2C2, -0x1C70, 0x1C70);
+    return true;
+}
+
+s32 func_80C2478C(DmGm* this, GlobalContext* globalCtx) {
+    if (this->unk_2B4 != NULL) {
+        func_80C2457C(this, globalCtx);
+        this->unk_2AE &= ~1;
+        this->unk_2AE |= 2;
+    } else if (this->unk_2AE & 2) {
+        this->unk_2AE &= ~2;
+        this->unk_2BE = 0;
+        this->unk_2C0 = 0;
+        this->unk_2C2 = 0;
+        this->unk_2C4 = 0;
+        this->unk_2BC = 20;
+    } else if (DECR(this->unk_2BC) == 0) {
+        this->unk_2AE |= 1;
+    }
+    return true;
+}
+
+Actor* func_80C24838(GlobalContext* globalCtx) {
+    Actor* tempActor;
+    Actor* foundActor = NULL;
+
+    while (true) {
+        foundActor = SubS_FindActor(globalCtx, foundActor, ACTORCAT_NPC, ACTOR_DM_AH);
+
+        if ((foundActor == NULL) || (foundActor->update != NULL)) {
+            break;
+        }
+
+        tempActor = foundActor->next;
+        if (tempActor == NULL || NULL) {
```

Comment:
```suggestion
        if (tempActor == NULL || false) {
```
Since it is just a boolean value.